src/crimson/net/SocketMessenger.cc

```diff
@@ -134,6 +134,8 @@ seastar::future<> SocketMessenger::start(Dispatcher *disp) {
 crimson::net::ConnectionRef
 SocketMessenger::connect(const entity_addr_t& peer_addr, const entity_name_t& peer_name)
 {
+  if (stopping)
+    return {};
```

Comment:
It would be better to throw an explicit error here to tell user that the messenger is already shutdown, which could be defined in `crimson::net::error`.